sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -82,17 +93,31 @@
         } else {
             String authorityUrl = options.getAuthorityHost().replaceAll("/+$", "") + "/organizations/" + tenantId;
             PublicClientApplication.Builder publicClientApplicationBuilder = PublicClientApplication.builder(clientId);
-            if (options.getHttpPipeline() != null) {
-                publicClientApplicationBuilder = publicClientApplicationBuilder
-                        .httpClient(new HttpPipelineAdapter(options.getHttpPipeline()));
-            }
             try {
                 publicClientApplicationBuilder = publicClientApplicationBuilder.authority(authorityUrl);
             } catch (MalformedURLException e) {
                 throw logger.logExceptionAsWarning(new IllegalStateException(e));
             }
-            if (options.getProxyOptions() != null) {
-                publicClientApplicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
+
+            // If user supplies the pipeline, then it should override all other properties
+            // as they should directly be set on the pipeline.
+            HttpPipeline httpPipeline = options.getHttpPipeline();
+            if (httpPipeline != null) {
+                httpPipelineAdapter = new HttpPipelineAdapter(httpPipeline);
+                publicClientApplicationBuilder.httpClient(httpPipelineAdapter);
+            } else {
+                // If http client is set on the credential, then it should override the proxy options if any configured.
+                HttpClient httpClient = options.getHttpClient();
+                if (httpClient != null) {
+                    httpPipelineAdapter = new HttpPipelineAdapter(setupPipeline(httpClient));
+                    publicClientApplicationBuilder.httpClient(httpPipelineAdapter);
+                } else if (options.getProxyOptions() != null) {
+                    publicClientApplicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.getProxyOptions()));
+                } else {
+                    //Http Client is null, proxy options are not set, use the default client and build the pipeline.
+                    httpPipelineAdapter = new HttpPipelineAdapter(setupPipeline(httpClient));
```

Comment:
Instead of passing `httpClient` here, you can as well use `HttpClient.createDefault()` since you know `httpClient` is null.